### PR TITLE
[export] fix loss_output in joint graph signature

### PR DESCRIPTION
**Pull request by pianpwk**
Summary: joint-graph export is marking all outputs as LOSS_OUTPUT, fix so it marks only the correct one

Test Plan: test_experimental

Differential Revision: D66117412

### CODE REVIEW

File: torch/export/graph_signature.py [hunk 1]
```diff
@@ -283,11 +283,15 @@ def user_inputs(self) -> Collection[Union[int, float, bool, None, str]]:
...
     # Graph node names of pytree-flattened outputs of original program
+    # For joint-graph purposes, will include the loss output.
     @property
     def user_outputs(self) -> Collection[Union[int, float, bool, None, str]]:
         user_outputs: List[Union[int, float, bool, None, str]] = []
         for s in self.output_specs:
-            if s.kind != OutputKind.USER_OUTPUT:
+            if s.kind not in [
+                OutputKind.USER_OUTPUT,
+                OutputKind.LOSS_OUTPUT,
```
Comment (ydwu4) [inline torch/export/graph_signature.py:293]: Why we need changing this?

File: test/export/test_experimental.py [hunk 2]
```diff
@@ -263,6 +263,26 @@ def forward(self, x, labels):
...
+                x = x.sum()
+                if self.index == 0:
+                    return x, -x.detach()
+                else:
+                    return x.detach(), x
+
+        inputs = (torch.randn(4, 4),)
+        for i in [0, 1]:
+            ep = export_for_training(Foo(i), inputs)
+            ep0 = _export_forward_backward(ep, joint_loss_index=i)
```
Comment (ydwu4) [inline test/export/test_experimental.py:284]: also test the graph signatures's output spec?

**Review by JacobSzwejbka — APPROVED** [review]

File: torch/export/graph_signature.py (hunk 1 above)
Comment (pianpwk) [reply torch/export/graph_signature.py:293]: For verifier purposes this in a way includes the loss output: https://github.com/pytorch/pytorch/blob/d276688da6c1881d058670624fda4589f68b4571/torch/_export/verifier.py#L427

We don't have anything specific for training in the verifier yet.. ok maybe this is just a lazy change.
